### DISCUSSION TITLE
Make setting `magit-branch-read-upstream-first` to `nil` more convenient

### DISCUSSION THREAD

**Discussion by elibarzilay** [magit/magit#5083] [Ideas]
When `magit-branch-read-upstream-first` is `t`, it's nice to `b c` some `origin/foo` and get a default of `foo` for the local branch name.  The thing is that I lose this convenience if I set it to `nil`.

It would be nice to do the equivalent convenience in this direction too: I'd `b c` that `foo` branch, and when it asks for a starting point, it would suggest `origin/foo` if it exists, instead of the usual (and likely-to-be-wrong) default of `master`.

(I made sure that this is not done in the recent version, and no threads or issues mention `magit-branch-read-upstream-first`.)

**Comment by tarsius** [comment]
Have you tried using `magit-branch-checkout` (`b l`) instead? It's purpose is to checkout a local branch, and usually that means an existing local branch. But the completion candidates include remote branches, and if you select on of them, then it creates the corresponding local branch and checks that out.
